packages/spec-lib/src/decorators.ts

```diff
@@ -209,17 +225,37 @@ function resolveScenarioName(target: Operation | Interface | Namespace, name: st
   return names.join("_");
 }
 
+const ScenarioKey = Symbol("Scenario");
 export function isScenario(program: Program, target: Operation | Interface | Namespace): boolean {
-  return program.stateMap(SpecLibStateKeys.Scenario).has(target);
+  return program.stateMap(ScenarioKey).has(target);
 }
 
 export function getScenarioName(
   program: Program,
   target: Operation | Interface | Namespace,
 ): string | undefined {
-  const name = program.stateMap(SpecLibStateKeys.Scenario).get(target);
+  const name = program.stateMap(ScenarioKey).get(target);
   if (name === undefined) {
     return undefined;
   }
   return resolveScenarioName(target, name);
 }
+
+const SupportedBy = Symbol("SupportedBy");
```

Comment:
revert the support by changes, we decided that it was not a necessary decorator and should be removed

Reply:
I have removed all the code related to `supportedBy` decorator.